app/controllers/summaries_controller.rb

```diff
@@ -0,0 +1,26 @@
+class SummariesController < ApplicationController
+ def new
+    @topic = Topic.find(params[:topic_id])
+    @post = Post.find(params[:post_id])
+    @summary = Summary.new
+  end
+
+  def show
+    @post = Post.find(params[:post_id])
+    @summary = @post.summary
+  end
+  
+  def create
+    @summary = Summary.new(params.require(:summary).permit(:description,:post_id))
```

Comment:
This works, but there is a lot going on with this line.  The parameters could be extracted to a separate method or variable which provide safe / white listed parameters.